Library/Homebrew/tap_constants.rb

```diff
@@ -1,9 +1,11 @@
+# allows tap names to start with https://www.github.com/, which is then stripped off
+HOMEBREW_TAP_URL_REGEX = %r{(?:(?:(?:https?://)?(?:www\.)?)?github\.com/)?}
```

Comment:
What about HOMEBREW_TAP_GITHUB_URL_PREFIX? This file is `require`d in global.rb, so I wanted to keep the HOMEBREW_TAP_ prefix on the name.